.github/workflows/ui.yaml

```diff
@@ -0,0 +1,147 @@
+name: "UI Build"
+on:
+  push:
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]+'
+      - '[0-9]+.[0-9]+.[0-9]+-rc[0-9]+'
+    branches: [ $default-branch ]
+    # Path filters aren't evaluated for tags - https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore
+    paths:
+      - '.nvmrc'
+      - 'services/ui/**'
+  pull_request:
+    paths:
+      - '.nvmrc'
+      - 'services/ui/**'
+    branches: [ $default-branch ]
+env:
+  REGISTRY: ghcr.io/noaa-gsl/unified-graphics/ui
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'npm'
+          cache-dependency-path: 'services/ui/package-lock.json'
+      - name: Install dependencies
+        working-directory: services/ui
+        run: npm ci
+      - name: Lint
+        working-directory: services/ui
+        run: npm run lint
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'npm'
+          cache-dependency-path: 'services/ui/package-lock.json'
+      - name: Install dependencies
+        working-directory: services/ui
+        run: npm ci
+      - name: Check
+        working-directory: services/ui
+        run: npm run check
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'npm'
+          cache-dependency-path: 'services/ui/package-lock.json'
+      - name: Install dependencies
+        working-directory: services/ui
+        run: npm ci
+      - name: Test
+        working-directory: services/ui
+        run: |
+          npx playwright install-deps
+          npm run test
```

Comment:
This is not important, but as an FYI: `test` is a script that is special with NPM and you can just execute `npm test` instead of `npm run test`. (`start` is another special script, but we’re not using that.)

Reply:
Oh, cool. I didn't realize that shortcut was there so thanks for mentioning it. I'll update it.